src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public class JsonContent : HttpContent
+    {
+        internal const string JsonMediaType = "application/json";
+
+        private readonly byte[] _content;
+        private readonly int _offset;
+        private readonly int _count;
+
+        private static MediaTypeHeaderValue CreateMediaType(string mediaTypeAsString)
+        {
+            MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse(mediaTypeAsString);
+
+            // If the instantiated mediaType does not specify its CharSet, set UTF-8 by default.
+            if (mediaType.CharSet == null)
+            {
+                mediaType.CharSet = Encoding.UTF8.WebName;
+            }
+
+            return mediaType;
+        }
+
+        public Type ObjectType { get; }
+
+        public object? Value { get; }
+
+        public JsonContent(Type type, object? value, JsonSerializerOptions? options = null)
+            : this(type, value, CreateMediaType(JsonMediaType), options) { }
+
+        public JsonContent(Type type, object? value, string mediaType, JsonSerializerOptions? options = null)
+            : this(type, value, CreateMediaType(mediaType?? throw new ArgumentNullException(nameof(mediaType))), options) { }
+
+        public JsonContent(Type type, object? value, MediaTypeHeaderValue mediaType, JsonSerializerOptions? options = null)
+            : this(JsonSerializer.SerializeToUtf8Bytes(value, type, options), type, value, mediaType) { }
+
+        private JsonContent(byte[] content, Type type, object? value, MediaTypeHeaderValue mediaType)
+        {
+            if (mediaType == null)
```

Comment:
The Formatting API allows nulls in which case  it uses the specified MediaTypeFormatter (which we don't have).

IMO, we should throw `ArgumentNullException` for this case (and any other case where we take a `mediaType` arg.
Why would someone intentionally try to pass a null mediaType? it sounds like something that can silently fail.

IF we allow null, however, lets not forget to annotate the argument as nullable.